scripts/compare_kev_vulnerabilities.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+#
+# Copyright 2025 Canonical Ltd. All rights reserved.
+
+set -euo pipefail
+
+# This script compares known vulnerabilities from the CISA KEV list
+# with the vulnerabilities found in a vulnerability report file.
+# KEV = Known Exploited Vulnerabilities
+#
+# The KEV file is provided in JSON format from CISA at
+# https://www.cisa.gov/known-exploited-vulnerabilities-catalog
+#
+# The vulnerability report file should be in SARIF format, as this
+# provides interoperability across different scanners.
+# SARIF is just JSON with a specific schema.
+
+# USAGE: VULN_REPORT_FILE=my-trivy.sarif KNOWN_CVES_FILE=my-known-vulns.sarif ./compare_kev_vulnerabilities.sh
+
+# Always run from repo root, assuming we are in scripts/
+cd "$(dirname "$0")/.."
+
+KNOWN_CVES_FILE="${KNOWN_CVES_FILE:-kev.json}"
+
+# Ensure files exist
+if [ ! -f "$VULN_REPORT_FILE" ]; then
+  echo "Error: Vulnerability report file '$VULN_REPORT_FILE' not found."
+  exit 2
+fi
+
+if [ ! -f "$KNOWN_CVES_FILE" ]; then
+  echo "Error: Known CVEs file '$KNOWN_CVES_FILE' not found."
+  exit 2
+fi
+
+# Extract CVE IDs from known vulnerabilities list
+known_cves="$(jq -r '.vulnerabilities[].cveID' "$KNOWN_CVES_FILE" | sort -u)"
+report_cves="$(jq -r '.runs[].results[].ruleId' "$VULN_REPORT_FILE" | sort -u)"
```

Comment:
It seems strange we're making a script to compare the lists. Has this not been done before / is this the "correct" way?

Reply:
I looked into what others did,
- Anbox team made a simple script which I copied, see [here](https://github.com/canonical/ams/blob/e518782854c12977be04b8e6ffe4024fdb97566b/scripts/compare_kev_vulnerabilities.sh). I copied their script and improved it to use a sarif file for detected vulnerabilities so that we can re-use this for govuln checks and for Trivy scans.
- Canonical Contracts has [the following](https://github.com/canonical/contracts/tree/main/scripts/ssdlc/secscan) which is a more extensive set of Python scripts.
- Online there are some actions like [this one](https://github.com/advanced-security/dependabot-kev-action) that don't really fit our needs.